src/layout/embedding.rs

```diff
@@ -0,0 +1,732 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use hashbrown::{HashMap, HashSet};
+use indexmap::{IndexMap, IndexSet};
+use petgraph::prelude::*;
+use petgraph::visit::NodeIndexable;
+use petgraph::Directed;
+use rayon::prelude::*; // For par_sort
+use std::fmt::Debug;
+
+use crate::StablePyGraph;
+use rustworkx_core::connectivity::connected_components;
+use rustworkx_core::planar::lr_planar::{LRState, Sign};
+
+pub type Point = [f64; 2];
+
+#[derive(Debug)]
+pub struct CwCcw<T> {
+    cw: Option<T>,
+    ccw: Option<T>,
```

Comment:
Is `Option` the correct type here? These values will never be `None`. Creating an instance holding a `None` and later updating the weights with the correct values is redundant: store the final values right from the beginning!